python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import logging
+import cPickle
+
+class _BootstrapLogHandler(logging.StreamHandler):
+    """
+    Manually flushes emitted records for js to pickup.
+    """
+
+    def emit(self, record):
+        """
+        Forwards the record back to to js via the engine communicator.
+
+        :param record: The record to log.
+        """
+        super(_BootstrapLogHandler, self).emit(record)
+
+        # always flush to ensure its seen by the js process
```

Comment:
I've removed the entirety of the custom logging stuff. That had slowly been pushed out of the way as I developed this, and it turned out that there was only a single debug message being sent from the non-engine logger. No need to keep it as a result.